src/structures/CommandInteractionOptionResolver.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+const { TypeError } = require('../errors');
+
+/**
+ * A resolver for command interaction options.
+ */
+class CommandInteractionOptionResolver {
+  constructor(client, options) {
+    /**
+     * The client that instantiated this.
+     * @name CommandInteractionOptionResolver#client
+     * @type {Client}
+     * @readonly
+     */
+    Object.defineProperty(this, 'client', { value: client });
+
+    /**
+     * The interaction options array.
+     * @type {CommandInteractionOption[]}
+     * @private
+     */
+    this._options = options ?? [];
+  }
+
+  /**
+   * Gets an option by its name.
+   * @param {string} name The name of the option.
+   * @param {boolean} [required=false] Whether to throw an error if the option is not found.
+   * @returns {?CommandInteractionOption} The option, if found.
+   */
+  get(name, required = false) {
+    const option = this._options.find(opt => opt.name === name);
+    if (!option) {
+      if (required) {
+        throw new TypeError('COMMAND_INTERACTION_OPTION_NOT_FOUND', name);
+      }
+      return null;
+    }
+    return option;
+  }
+
+  /**
+   * Gets an option by name and property and checks its type.
+   * @param {string} name The name of the option.
+   * @param {ApplicationCommandOptionType[]} types The type of the option.
+   * @param {string[]} properties The property to check for for `required`.
```

Comment:
So is it a property or many properties?

Reply:
Many properties. Looks like that got outdated, shall fix